src/internal/_curryN.js

```diff
@@ -0,0 +1,38 @@
+var arity = require('../arity');
+
+
+/**
+ * Internal curryN function.
+ *
+ * @private
+ * @category Function
+ * @param {Number} length The arity of the curried function.
+ * @return {array} An array of arguments received thus far.
+ * @param {Function} fn The function to curry.
+ */
+module.exports = function _curryN(length, recieved, fn) {
+  return function() {
+    var combined = [];
+    var argsIdx = 0;
+    var left = length;
+    var combinedIdx = 0;
+    while (combinedIdx < recieved.length || argsIdx < arguments.length) {
+      var result;
+      var rec = recieved[combinedIdx];
+      var recNotPlaceholder = rec == null || rec['@@functional/placeholder'] !== true;
```

Comment:
> What about a helper function?

I assumed a helper function was off the table for performance reasons, but I'm far from a performance guru. :)

> (I can't believe I'm suggesting a String constant after railing against them over and over at work, but I think it makes sense here.)

I've been on your side of this debate many times myself. My argument has been that using `someModule.FOOBAR` (whose value is `'foobar'`) provides no benefit over using `'foobar'` directly. In this case, though, there would be a benefit: JSHint would tell us if we mistyped `placeholder`.